src/modal-header-colors.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Some some colors that can be used for the header
+ *
+ * @export
+ * @enum {number}
+ */
+export enum ModalHeaderColors {
```

Comment:
I have yet to write TypeScript, and am wondering why an enum of strings is preferred to a singleton object with the same property assignments.

Reply:
So this is actually not a great use case for an enum and I was considering removing it from this repo.. it's really just a few presets that we can use for the header colors. We're not restricting the `headerColor` that you can pass in since it's just a string in `ModalConfig.headerColor`.

Regarding your question about enum vs singleton object.. a better use case of an enum would be if we only allowed specific colored modals. The configuration would instead look like this:

```
class ModalConfig {
  headerColor: ModalHeaderColors = ModalHeaderColors.Green; // default value
}
...
const config = new ModalConfig();
config.headerColor = ModalHeaderColors.Blue;
config.headerColor = ModalHeaderColors.Purple; // compile error
```

Now we've restricted our options of what we are allowing `headerColor` to be. The Typescript compiler will now enforce the developer to only allow options that are in `ModalHeaderColors`.

Enums are really useful when you have a fixed set of options and you want to enforce developers to only use one of those options.

Reply:
I figured it had to do with enforcing one of a set of inputs. Does the same value enforcement occur when the enum is populated with integer values?

Reply:
Yeah. Actually enums are integers by default, but assigning string values to them turns them into a string. There's actually [quite a bit](https://www.typescriptlang.org/docs/handbook/enums.html) you can do with enums, even setting the values to callbacks.